blog/_src/posts/2017-08-24-closure-conversion-coyoneda.md

```diff
@@ -0,0 +1,294 @@
+M(B,B) \\) such that the \\(A, B\\) positions are related by some
+function \\(f : A \to B \\), then \\(pack_A(mA) = pack_B(mB)\\).
+More formally, this means that if you have a \\(m' : M(B,A) \\), then
+
+\\[ pack_B(M(B,f)(m')) = pack_A(M(f,A)(m'))\\]
+or in a point-free style:
+\\[ pack_B \circ M(B,f) = pack_A \circ M(f,A) : M(B,A) \to \exists X. M(X,X) \\]
+
+A function parameterized by types like \\(pack \\) that has this
+property is called a "co-wedge" from \\(M \\).
+
+A coend is a type \\(\exists M(X,X) \\) and a co-wedge \\(\forall
+A. pack_A : M(A,A) \to \exists X. M(X,X) \\) that are *universal*,
+i.e. any other family of maps \\(f_A : M(A,A) \to C\\) factors through
+\\(pack \\). This gives us the sytnax for existential elimination.
+
+If you are familiar with parametricity, it is a good exercise to see
+why the usual condition for invariance wrt all *relations* implies
+that a parametric \\(pack, \exists X. M(X,X) \\) will form a cowedge.
+It seems that in general it would not be a universal co-wedge because
+a parametric exists is invariant under all relations and there are
+many relations that don't act like functions.
+
+### Presheaves
+
+Next, a presheaf is just another word for functor \\( Q : C^{op} \to
+\Set \\).
+Think of this as a set that is parameterised by a type of "inputs", so
+if you have a map in \\(C, f : A \to B\\) you get a function \\(Q(f) :
+Q(B) \to Q(A) \\) that "preprocesses" the inputs using \\(f
+\\). Functoriality ensures that preprocessing with the identity is
+just the identity and that composition or preprocessers is the
+preprocessor from the composite function.
+
+So the informal explanation of the coYoneda lemma is that for any
+presheaf \\(Q \\), if we have an \\( \exists X. (A \to X) \times Q(X)
+\\), then since we can't inspect the \\(X \\) in any way, all we can
+really do is compose the \\(Q(X) \\) with the preprocesser from the
+function \\(A \to X \\), giving us a \\(Q(A) \\).
+
+### Enriched Categories and Enriched CoYoneda
+But there's a gap from here to applying this to a programming
+language, the coYoneda lemma as presented says that \\(Q(A) \\) and
+\\(\exists B. (A \to B) \times Q(B) \\) isomorphic as *sets*, but we
+wanted an isomorphism of *types* in our programming language.  We can
+reconcile this by considering *enriched* category theory and the
+*enriched* coYoneda lemma.  Let \\(V \\) be a category, then if \\( V
+\\) is sufficiently like the category of sets, then we can do a lot of
+category theory by replacing the word "set" with "object of \\(V \\)".
+
+Specifically, a \\(V \\)-enriched category (or just \\(V\\)-category)
+has a set of objects \\(Ob \\), but for each pair of objects \\(A,B
+\in Ob \\) we get a \\( V\\)-object \\(\Hom(A,B) \\) of morphisms from
+\\(A \\) to \\( B \\).  If \\(V \\) is a closed category, we can see
+\\(V \\) *itself* as a \\(V\\)-enriched category with the same objects
+and just making \\(\Hom(A,B) = A \to B \\).
```

Comment:
`A \to B` here is weird to me

at first I thought this was a typo, using an arrow here
then I looked on nLab and see the notation `hom(A,B)` and `[A,B]`

suggestion: "we can see `\\(V \\)` *itself* as a V-category with the same objects because V contains an internal hom object corresponding to each `\Hom(A,B)`"